src/Sylius/Bundle/CoreBundle/Checkout/Step/PurchaseStep.php

```diff
@@ -55,21 +54,23 @@ public function forwardAction(ProcessContextInterface $context)
         $this->getHttpRequestVerifier()->invalidate($token);
 
         $status = new GetStatus($token);
-        $this->getPayum()->getPayment($token->getPaymentName())->execute($status);
+        $this->getPayum()->getGateway($token->getGatewayName())->execute($status);
 
         /** @var $payment PaymentInterface */
         $payment = $status->getFirstModel();
         $order = $payment->getOrder();
 
+        // Should it be here??
```

Comment:
@pjedrzejewski looks strange to me, the purchase is already done, but we dispatch initialize event, is it ok?